components/board/customlinux/board.go

```diff
@@ -87,13 +87,13 @@ func parseRawPinData(pinData []byte, filePath string) ([]genericlinux.PinDefinit
 	for name, pin := range parsedPinData.Pins {
 		err = multierr.Combine(err, pin.Validate(filePath))
 
-		// Until we get hardware PWM working reliably on lots of boards (likely by being able to
-		// set the pin mode directly), we disable all hardware PWM on these boards.
-		// The range operator makes a copy of the value, so to change the original, we need to look
-		// it up in the map again.
-		// TODO(RSDK-5105): Undo this when we're ready to support hardware PWM.
-		parsedPinData.Pins[name].PwmChipSysfsDir = ""
-		parsedPinData.Pins[name].PwmID = -1
+		// Until we can reliably switch between gpio and pwm on lots of boards, pins that have
+		// hardware pwm enabled will be hardware pwm only. Disabling gpio functianality on these
+		// pins.
+		if parsedPinData.Pins[name].PwmChipSysfsDir != "" {
```

Comment:
I'd also check that the line number is non-negative. Some people might be lazy and disable a pin by leaving the directory as-is and setting the line to a negative value.